src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+
+    // This holds the FileList
+    const fileInput = ref([]) as any
+    // To clear the input value after reset
+    const fileInputKey = ref(0)
+    // File fakepath
+    const fileValue = ref('') as any
+    // Array to store the previously selected FileList when user clicks reopen the file uploader and clicks on Cancel
+    const fileClone = ref([]) as any
+
+    const onFileChange = (evt: any): void => {
+      fileInput.value = evt.target?.files
+      fileValue.value = fileInput?.value[0]?.name
+
+      const fileSize = fileInput?.value[0]?.size
+
+      if (props.appearance === 'file') {
+        hasUploadError.value = fileSize > maximumFileSize.value
+        if (hasUploadError.value) fileInputKey.value++
+        emit('file-upload-error', fileInput.value)
+      } else {
+        hasUploadError.value = fileSize > maximumFileSize.value
+        if (hasUploadError.value) fileInputKey.value++
+        emit('file-upload-error', fileInput.value)
+      }
+
+      const inputElem = document.getElementById(customInputId.value) as HTMLInputElement
+
+      if (fileSize) {
+        fileClone.value.push(fileInput.value)
+      } else {
+        inputElem.files = fileClone.value[fileClone.value.length - 1]
+        fileInput.value = inputElem.files
+        if (inputElem.files) {
+          fileValue.value = inputElem.files[inputElem.files.length - 1].name
+        }
+      }
+      emit('file-added', fileInput.value)
+    }
+
+    // When KButton for Select file is clicked
+    const updateFile = (): void => {
+      const inputEl = document.getElementById(customInputId.value)
+      if (inputEl) {
+      // Simulate button click to trigger input click
+        inputEl.click()
+      }
+    }
+
+    // When Cancel button is clicked
+    const resetInput = (): void => {
+      fileInput.value = ''
+      fileValue.value = ''
+      fileClone.value = []
+      fileInputKey.value++
+      hasUploadError.value = false
+
+      emit('file-removed')
+    }
+
+    return {
+      fileInput,
+      customInputId,
+      resetInput,
+      onFileChange,
+      fileInputKey,
+      fileValue,
+      updateFile,
+      hasUploadError,
+      fileClone,
+      maximumFileSize,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+.k-file-upload {
+  position: relative;
+}
+
+.k-file-upload .k-file-upload-btn.k-button {
+  position: absolute;
+  right: 15px;
+  top: 35px;
+  border-radius: 100px;
+  height: 29px;
+}
+
+// To hide the thumbnail that appears in Safari after uploading a file
+.k-file-upload :deep(.k-input-wrapper) input[type="file"]::-webkit-file-upload-button {
+  position: absolute;
+  min-width: 100%;
+  min-height: 100%;
+  opacity: 0;
+  cursor: inherit;
+}
+
+.k-file-upload :deep(.k-input-wrapper) input[type="file"],
+.k-file-upload :deep(.k-input-wrapper) input[type="file"].image-upload {
+  color:transparent;
+}
+
+#pseudoCancel {
```

Comment:
Please use a class instead of an id.
```suggestion
.pseudoCancel {
```